### PR TITLE
made PF_SHARED_CONTENT_FOLDER available to vm startup scripts

### PR DESCRIPTION
**Pull request by brianwp3000**
This fix is specifically for the Watson team. Previously we gave their startup script the PF_SHARED_CONTENT_FOLDER_VM variable so that their agent process knows where to find crash dumps, but it turns out their script also needs the PF_SHARED_CONTENT_FOLDER variable so that they can set the linux kernel core path correctly for container-based servers to place their dumps in the shared content folder.

So for example:
PF_SHARED_CONTENT_FOLDER_VM = /mnt/GameSharedContent <- this folder exists on the VM
PF_SHARED_CONTENT_FOLDER (on a process based server) = /mnt/GameSharedContent
PF_SHARED_CONTENT_FOLDER (on a container based server) = /data/GameSharedContent <- this folder exists only exists inside a container, and is where PF_SHARED_CONTENT_FOLDER_VM is mounted
Their startup script tells the Watson agent to look for crash dumps in PF_SHARED_CONTENT_FOLDER_VM. The linux kernel core path tells linux where to place crash dumps, so they set it to PF_SHARED_CONTENT_FOLDER_VM. Turns out this doesn't work for containers--when a process in a container crashed, it tried placing the crash dump into /mnt/GameSharedContent which is a path that doesn't exist inside the container.
But they can get it to work if they set the linux kernel core path to PF_SHARED_CONTENT_FOLDER--that way when a process inside a container crashed, it got placed in /data/GameSharedContent (which is mounted to /mnt/GameSharedContent like we want). And for process based servers PF_SHARED_CONTENT_FOLDER is set to /mnt/GameSharedContent like we want. 

tl;dr PF_SHARED_CONTENT_FOLDER_VM is the true path to the shared content folder on the VM, while PF_SHARED_CONTENT_FOLDER is the path where a server *thinks* the shared content folder is. Turns out both are useful for VM startup scripts to have

### CODE REVIEW

File: VmAgent.Core/VmConfiguration.cs [hunk 1]
```diff
@@ -42,6 +42,11 @@ public class VmConfiguration
         // should use `PF_SHARED_CONTENT_FOLDER` instead (which points to wherever the folder is mounted in the container)
         public const string SharedContentFolderVmVariable = "PF_SHARED_CONTENT_FOLDER_VM";
 
+        // Used by the games to share user generated content (and other files that are downloaded once, used multiple times).
```
Comment (dgkanatsios) [inline VmAgent.Core/VmConfiguration.cs:45]: can you add here the great explanation you put on the PR description? This will help the reader understand why we need two env variables.

File: VmAgent.Core/VmConfiguration.cs [hunk 2]
```diff
@@ -42,6 +42,11 @@ public class VmConfiguration
...
+        // Also available to VM-level scripts for special cases (e.g. Watson needs to set the Kernel core dump path to this path
+        // so that container-based servers place their dump files in the path available to the container)
+        private const string SharedContentFolderEnvVariable = "PF_SHARED_CONTENT_FOLDER";
```
Comment (dgkanatsios) [inline VmAgent.Core/VmConfiguration.cs:48]: should we also add some tests to make sure the correct set of env variables are being set?

**Review by jotavalenciano — APPROVED** [review]

File: VmAgent.Core/VmConfiguration.cs (hunk 1 above)
Comment (brianwp3000) [reply VmAgent.Core/VmConfiguration.cs:45]: I expanded the comment (but it's still not as long as the novel i wrote for the PR description)

File: VmAgent.Core/VmConfiguration.cs (hunk 2 above)
Comment (brianwp3000) [reply VmAgent.Core/VmConfiguration.cs]: Added unit tests

**Review by dgkanatsios — APPROVED** [review]
